### ISSUE TITLE
The Solr server could not be reached.

### ISSUE THREAD

**Issue by osopolar** [lando/lando#1047]
I am using lando Version: v3.0.0-beta.47 on macOS 10.13.5 with below .lando.yml `lando start` looks fine, BOOMSHAKALAKA!!!:

```yml
name: example
recipe: drupal7
config:
  php: '7.1'
  via: apache
  webroot: .
  drush: composer
  database: mysql:5.6
  xdebug: false
  conf:
    php: sites/all/etc/lando/dev.php.ini
    server: sites/all/etc/lando/dev.httpd-ssl.con
    database: sites/all/etc/lando/dev.mysq
proxy:
  mailhog:
    - mail.example.lndo.site
  solr:
    - solr.example.lndo.site:8983
services:
  appserver:
    composer:
      davereid/drush-patchfile: dev-master
    build:
      - "cd $LANDO_MOUNT && composer install"
  mailhog:
    type: mailhog
    hogfrom:
      - appserver
    portforward: 1026
  solr:
    type: solr:7.1
    portforward: 9999
    core: example
    config:
      conf: sites/all/etc/lando/dev.solr-conf
tooling:
  drush:
    service: appserver
    cmd:
      - "/app/vendor/bin/drush"
  phpcs:
    service: appserver
    cmd: "/app/vendor/bin/phpcs --standard=Drupal,DrupalPractice"
    options:
    description: Run phpcs for given folder or file.
```

I use Drupal 7 versions of Search API (7.x-1.24) and Serach API Solr (7.x-1.12+7-dev), with the modules default Solr 7.x configuration (https://cgit.drupalcode.org/search_api_solr/tree/solr-conf/7.x?h=7.x-1.x). I only modified solr.contrib.dir in the [solrcore.properties](https://cgit.drupalcode.org/search_api_solr/tree/solr-conf/7.x/solrcore.properties?h=7.x-1.x) file:

```diff
- solr.contrib.dir=../../../contrib
+ solr.contrib.dir=../../../../contrib
```

I can reach the admin interface by the urls http://solr.example.lndo.site and https://solr.example.lndo.site

I configured the Search API Solr Server following the instructions from https://www.drupal.org/node/1999386:

* HTTP protocol: `http`
* Solr host: `solr.example.lndo.site`
* Solr port: `8983`
* Solr path `/solr`

This results in the Solr server URI: `http://solr.example.lndo.site:8983/solr` (can't open that URL from my browser) and gives the connection status: The Solr server could not be reached. Further data is therefore unavailable.

Changing Solr port to `9999` results in the Solr server URI `http://solr.example.lndo.site:9999/solr`, which I also can open in may local web-browser, but the connection status is still `The Solr server could not be reached`.

Changing Solr host to `localhost` don't work eighter. I tried a bunch of other Solr host and port combinations but nothing worked so far.

So what am I missing, what is the problem? What does it need that the appserver cann access the solr server?

Output from `lando info`:

```json
{
  "solr": {
    "type": "solr",
    "version": "7.1",
    "hostnames": [
      "solr",
      "solr.example.internal"
    ],
    "internal_connection": {
      "core": "example",
      "host": "solr",
      "port": 8983
    },
    "external_connection": {
      "core": "example",
      "host": "localhost",
      "port": 9999
    },
    "config": {
      "conf": "sites/all/etc/lando/dev.solr-conf"
    },
    "urls": [
      "http://solr.example.lndo.site",
      "https://solr.example.lndo.site"
    ]
  }
}
```

**Comment by dustinleblanc** [comment]
@osopolar I believe you want to use the 'internal connection' for solr, never localhost, and not the proxy domain. Set Solr host to `solr` as that is the hostname that docker will assign to the container. You have to remember that each service is a container inside a network and that docker by default assigns hostnames to each other service in that network. The most reliable address for any service is going to be that internal hostname (when making a request from another service). Using the proxy domain seems to sometimes work for some people, but I never trust it, especially since colleagues can globally reconfigure their proxy domain, causing your app to fail when they load it up.

Hope that helps! I am going to close this as I assume that will get you what you need, but feel free to let us know if it doesn't help, thanks!

**Comment by osopolar** [comment]
Thanks for answering, @dustinleblanc. I also remembered that proxy domain probably wont work reliable. I also tried host names `solr` and `solr.example.internal`, but that didn't work nigher.

I guess the problem is somewhere else. I connected to the appserver with `lando ssh appserver` and did a `wget http://solr:8983/solr/`. That works so far, the result is the same as if I run `wget http://solr.example.lndo.site/solr/#/`. So maybe it's a problem with Search API Solr module. I will have a look on that and post the results later.

**Comment by osopolar** [comment]
I better should have used the [Installing Solr 6.x](https://www.drupal.org/node/2794809) documentation, although installing Solr 7.x instead of [Configuring the Search API server](https://www.drupal.org/node/1999386), as in the first one it was explained:

> ... you will just have to append the core name you've chosen to the path. E.g., for the above instructions, use "/solr/drupal" instead of "/solr".

That was not part of the documentation I used (therefore I overlooked it). I already wondered how Search API would know which core to use but thought that this needs to be configured somewhere else.

See screenshot for the solr search server configuration. ... And while preparing that screenshot I also found that adding the core to the path was part of the field description 🤦

![solrservice](https://user-images.githubusercontent.com/1523215/41689776-c1423f86-74c8-11e8-8711-b55d751e2e8c.png)
